labai-jsr305-jaxb-plugin/src/main/java/com/sun/tools/xjc/addon/labai/JaxbNonnullPlugin.java

```diff
@@ -218,13 +219,28 @@ private void processNonnull(ClassOutline co, JFieldVar field) {
         }
     }
 
+    private boolean matchByInstrospection(String name, JMethod method) {
+        return method.body()
+                .getContents()
+                .stream()
+                .findFirst()
+                .filter(c -> c instanceof JStatement)
+                .map(c -> {
+                    StringWriter writer = new StringWriter();
+                    JFormatter f = new JFormatter(writer);
+                    ((JStatement) c).state(f);
+                    return writer.toString();
+                })
+                .filter(statement -> statement.contains("return " + name + ";"))
+                .isPresent();
+    }
     private JMethod getGetter(ClassOutline co, JFieldVar field) {
         String capitalizedName = field.name().substring(0, 1).toUpperCase() + field.name().substring(1);
         String getterName = "get" + capitalizedName;
         String booleanName = "is" + capitalizedName;
         // todo make non n^2 algorithm
         JMethod getter = co.implClass.methods().stream()
-                .filter(it -> getterName.equals(it.name()) || booleanName.equals(it.name()))
+                .filter(it -> getterName.equals(it.name()) || booleanName.equals(it.name()) || matchByInstrospection(field.name(), it))
```

Comment:
idea is good, just have a feeling, that doing calculation on filtering may have negative impact on performance